articles/tech-resources/ios-universal-links-self-serve.md

```diff
@@ -0,0 +1,109 @@
+---
+title: "Using iOS Universal Links with SparkPost"
+description: "A guide to including iOS universal links in your SparkPost-delivered email"
+---
+
+## Introduction
+
+Universal links are a mechanism in iOS 9 and later that allows you to link deeply into your iOS apps from web pages and HTML email. To use universal links, you must publish the URLs you'd like your iOS app to handle and then add support for them into the app itself. When your users receive an email containing a universal link, iOS then recognises the domain and path of the URL and triggers your app. As a result, it's important to keep the URLs in your HTML emails intact to ensure your universal links function correctly.
+
+This article describes the various ways you can use universal links with email sent through your SparkPost account.
+
+[Apple's Universal Links](https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/UniversalLinks.html) documentation explains the iOS app mechanisms in detail. As such, we will not restate those details here.
+
+## Prerequisites
+
+- an iOS app with universal link support
+- an apple-app-site-association file hosted on your own domain
+
+#### Sidebar: SparkPost Click Tracking
+
+In the discussion below, it might be useful to understand how the SparkPost click tracking feature works. In essence, when the feature is enabled and SparkPost encounters an anchor tag in an HTML email, it will replace the `href` attribute with a new URL pointing to it's click tracking service. When your recipient clicks that link, the click tracker receives the request, records the "click" and redirects the recipient to your original URL.
+
+You can customize the domain SparkPost uses in your tracked links by setting up a [custom tracking domain](https://app.sparkpost.com/account/tracking-domains). This is useful from a branding and reputation standpoint and it also offers a mechanism for including universal links in your HTML email.
+
+You can also have SparkPost include a specific string in the path part of a tracked URL by setting the `data-msys-sublink` attribute: `<a href="http://my.universal-link.example.com" data-msys-sublink="open-in-app">Open in app</a>`. This will cause the tracked URL path to start with `/f/open-in-app/` which simplifies writing your `apple-app-site-association` file.
```

Comment:
You're right, there is a missing step.

**At app installation time**
During installation, the app says to iOS: "Open me when you see links to this domain". iOS then looks for `apple-app-site-association` on that domain. By publishing the file, you acknowledge that the app is allowed to intercept links intended for the domain. The file also provides the other half of the universal link spec: a set of URL patterns.

Note: `apple-app-site-association` is served over HTTPS _at installation time_.

At this point, iOS has a list of universal link domains and URL patterns for your app.

**SparkPost account setup**
SparkPost users must ensure that universal links in their SparkPost-sent email use the correct domain (a custom tracking domain) and path (using a custom sublink) to match the app configuration.

Setting up the custom tracking domain requires a CNAME to `spgo.io` which makes it difficult to serve `apple-app-site-association` or anything else on that domain. The solution is to publish the CNAME, verify the tracking domain _and then remove the CNAME_.

**When the user clicks on a link**
When a user clicks on a link in an HTML email (or a browser), iOS checks it's list of universal link patterns and chooses whether to open an app or just follow the link like normal.

It passes a "universal link clicked" event to the app, which then swaps the custom tracking domain for `spgo.io` before making an HTTP request to let SparkPost register the "click".

(For reference, I chose to avoid re-iterating how Apple implements Universal Links to avoid unnecessary sprawl. Check out our [Enterprise support article](https://www.sparkpost.com/docs/tech-resources/ios-universal-links/) and competitors' articles to see what I mean. I made the assumption that users reading this will have a grounding in iOS app dev.)